diff --git configure configure
new file mode 100755
index 000000000..82c4d5042
--- /dev/null
+++ configure
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+# We have a prepared configuration, copy it to where it is expected.
+# If no configuration is present, then do nothing, we will likely generate it.
+if [ -n "$CONFIG" ]; then
+  CFLAGS_ESC=$(printf "%s\n" "${CFLAGS}" | sed 's/[\&/]/\\&/g')
+  LDFLAGS_ESC=$(printf "%s\n" "${LDFLAGS}" | sed 's/[\&/]/\\&/g')
+  sed --regexp-extended \
+    --expression "s/^CONFIG_EXTRA_CFLAGS=\"(.*)\"$/CONFIG_EXTRA_CFLAGS=\"${CFLAGS_ESC}\"/" \
+    --expression "s/^CONFIG_EXTRA_LDFLAGS=\"(.*)\"$/CONFIG_EXTRA_LDFLAGS=\"${LDFLAGS_ESC}\"/" \
+    "${CONFIG}" > .config
+fi
diff --git scripts/trylink scripts/trylink
index 2255deee7..583ed4af7 100755
--- scripts/trylink
+++ scripts/trylink
@@ -96,7 +96,7 @@ fi
 START_GROUP="-Wl,--start-group"
 END_GROUP="-Wl,--end-group"
 INFO_OPTS() {
-	echo "-Wl,--warn-common -Wl,-Map,$EXE.map -Wl,--verbose"
+	echo "-Wl,-Map,$EXE.map -Wl,--verbose"
 }
 
 # gold may not support --sort-common (yet)
@@ -320,7 +320,6 @@ int main(int argc, char **argv)
 		$SORT_SECTION \
 		$GC_SECTIONS \
 		-L"$sharedlib_dir" -lbusybox \
-		-Wl,--warn-common \
 	|| {
 	    echo "Linking $EXE failed"
 	    cat $EXE.out
